test/extensions/filters/http/buffer/buffer_filter_integration_test.cc

```diff
@@ -121,5 +148,41 @@ TEST_P(BufferIntegrationTest, RouteOverride) {
   EXPECT_STREQ("200", response->headers().Status()->value().c_str());
 }
 
-} // namespace
+TEST_P(BufferIntegrationTest, RequestPathTimesOutInBuffer) {
+  std::chrono::milliseconds test_connection_initiation_timeout(1250);
+  std::chrono::seconds buffer_timeout(1);
+
+  ConfigHelper::HttpModifierFunction mod = overrideConfigBufferTimeout(buffer_timeout);
+  config_helper_.addConfigModifier(mod);
+  config_helper_.addFilter(ConfigHelper::SMALL_BUFFER_FILTER);
+
+  setupRequestTimeoutTest();
+  EXPECT_EQ(getBufferTimeoutCount(), 0);
+
+  AssertionResult result = fake_upstreams_[0]->waitForHttpConnection(
+      *dispatcher_, fake_upstream_connection_, test_connection_initiation_timeout);
+
+  EXPECT_FALSE(result);
+  EXPECT_STREQ("408", response_->headers().Status()->value().c_str());
+  EXPECT_EQ(getBufferTimeoutCount(), 1);
+}
+
+TEST_P(BufferIntegrationTest, RequestPathTimesntOutInBuffer) {
+  std::chrono::milliseconds test_connection_initiation_timeout(1250);
+  std::chrono::seconds buffer_timeout(5); // Greater than test timeout
```

Comment:
suggestion: "Greater than test timeout above" or "Greater than test connection initiation timeout", just to make completely clear that you're referring to the other value you've declared.